src/OrleansServiceBus/Providers/Streams/EventHub/EventHubAdapterFactory.cs

```diff
@@ -102,10 +103,20 @@ public class EventHubAdapterFactory : IQueueAdapterFactory, IQueueAdapter, IQueu
 
         /// <summary>
         /// Create a receiver monitor to report performance metrics.
-        ///   Arguments are EventHub path, EventHub partition, and logger.
-        ///   Factory funciton should return an IEventHubReceiverMonitor.
+        /// Factory funciton should return an IEventHubReceiverMonitor.
         /// </summary>
-        protected Func<string, string, Logger, IEventHubReceiverMonitor> ReceiverMonitorFactory { get; set; }
+        protected Func<EventHubReceiverMonitorDimentions, Logger, IEventHubReceiverMonitor> ReceiverMonitorFactory { get; set; }
+        /// <summary>
+        /// Create a cache monitor to report performance metrics.
+        /// Factory funciton should return an ICacheMonitor.
+        /// </summary>
+        protected Func<EventHubCacheMonitorDimentions, Logger, ICacheMonitor> CacheMonitorFactory;
```

Comment:
We've delegated cache creation responsibility to the IEventHubQueueCacheFactory, which is also responsible for the buffer pool.  Given that, I think customization of the monitoring for these systems should also be delegated to the IEventHubQueueCacheFactory.